### PR TITLE
Query components unsafe lifecycle methods

### PR DESCRIPTION
**Pull request by boyskila**
**What this PR does / why we need it**:

   Refactor ```QueryEditor.tsx and QueryField.tsx``` deprecated lifecycle methods.
   In React 16.12 old names continue to work but you will now see a warning when using any of the old names.
This PR refactor only those related to the query components ```QueryEditor.tsx QueryField.tsx``` as it's suggested [here](https://github.com/grafana/grafana/issues/18503#issuecomment-566002113)

### CODE REVIEW

**Comment by boyskila** [comment]
@dprokop can you please review? Thanks!

File: public/app/plugins/datasource/cloudwatch/datasource.ts
Comment (hugohaggmark) [inline public/app/plugins/datasource/cloudwatch/datasource.ts]: Would this improve readability?
```suggestion
          item.expression?.length > 0)
```

File: packages/grafana-ui/src/components/QueryField/QueryField.tsx
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/QueryField/QueryField.tsx]: Do we really need to call `this.onChange` here or do we just need to trigger the bogus edit? wdyt @davkal @aocenas @kaydelaney

**Review by hugohaggmark — COMMENTED** [review]
Looks great! Left some minor comments that you should take a look at and waiting for comments/approval from @davkal @dprokop

File: packages/grafana-ui/src/components/QueryField/QueryField.tsx
Comment (aocenas) [reply packages/grafana-ui/src/components/QueryField/QueryField.tsx]: Honestly no idea, moving this to componentDidUpdate may have changed the need for this. @davkal do you know what was this solving/fixing?

File: public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx
Comment (dprokop) [inline public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx]: I like this approach, any comments on this @davkal ?

File: packages/grafana-ui/src/components/QueryField/QueryField.test.tsx
Comment (dprokop) [inline packages/grafana-ui/src/components/QueryField/QueryField.test.tsx]: Unnecessary line repeat?

File: public/app/plugins/datasource/cloudwatch/datasource.ts
Comment (boyskila) [reply public/app/plugins/datasource/cloudwatch/datasource.ts]: Thanks, looks better!

File: packages/grafana-ui/src/components/QueryField/QueryField.test.tsx
Comment (boyskila) [reply packages/grafana-ui/src/components/QueryField/QueryField.test.tsx]: No, actually is testing whether the logic inside ```if``` is executed only once, although I'm not sure how meaningful it is. Maybe in the future, if someone changes the logic by mistake this test will complain.

**Comment by davkal** [comment]
https://github.com/grafana/grafana/pull/20936/files#diff-62980d881b3e4c5cfc60d68e1756c3bfL237 has removed (AFAIK) the only use of an async syntax configuration. With this gone, `QueryField` could remove `syntaxLoaded` and all logic around it. 

@alexanderzobnin @dprokop @ryantxu can you confirm that no other plugin's query field uses `syntaxLoaded` in that way?

**Comment by alexanderzobnin** [comment]
@davkal Splunk plugin uses `syntaxLoaded` prop, so I'll test it out with new changes.

**Comment by boyskila** [comment]
@alexanderzobnin did you manage to test the Splunk plugin with the latest changes?

**Comment by alexanderzobnin** [comment]
@boyskila I tested it out against Splunk plugin and its query editor works well :+1:

**Comment by boyskila** [comment]
> @boyskila I tested it out against Splunk plugin and its query editor works well 👍

Thanks! @hugohaggmark @dprokop @davkal @aocenas I'm going to resolve the conflicts and I think is ready to be merged.

File: public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx
Comment (aocenas) [reply public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx]: So seems like this makes some tests fail, I think the issue is that this creates copy of props and `this.props` is not normalised while we pass that at least https://github.com/grafana/grafana/pull/21163/files#diff-3d15a6bca92bf6d3cab962e101acb86eR67.

**Review by aocenas — CHANGES_REQUESTED** [review]
@boyskila There seems to be a small issue with the props normalisation which is probably the reason for the tests failing. Can you take a look at that? If not, it's no problem we can take it over.

**Comment by boyskila** [comment]
> @boyskila There seems to be a small issue with the props normalisation which is probably the reason for the tests failing. Can you take a look at that? If not, it's no problem we can take it over.

I'll check, sure

**Comment by boyskila** [comment]
@aocenas Hello! Good catch! Yes, indeed my changes introduced this bug but at my very first commit, this issue wasn't captured. Anyways, passing the normalized query to the QueryFieldsEditor solved the issue.

**Comment by ivanahuckova** [comment]
@aocenas strictNullChecks are fixed, circle ci builds are passing. However the license/cla still needs to be signed. 

Regarding my commits: 

- [merge the master](https://github.com/grafana/grafana/pull/21163/commits/f2e4c85c7c7adff3e63c60a23399817b691c7b0e), so the new build pipeline is run
- [remove strictNullCheck errors] (https://github.com/grafana/grafana/pull/21163/commits/73102c86c55f3a58debbb2236409071257b1baed), but unfortunately not all of them
- [remove strictNullCheck errors](https://github.com/grafana/grafana/pull/21163/commits/a3efb0f3760bbc716087741229a1d736f83f0527) (not sure why using  `(item.expression && item.expression.length > 0))` instead of `item.expression?.length > 0` introduces strictNullCheck errors. Still a mystery to me. 
- [merge updated master](https://github.com/grafana/grafana/pull/21163/commits/36fc4984d6d1863992ad1e339ef246fc59d46b3d) as there was an update to docs-related error that was filing circle ci builds.

**Review by ivanahuckova — APPROVED** [review]

**Review by aocenas — APPROVED** [review]

**Comment by ivanahuckova** [comment]
Thank you @boyskila for this PR. 🙏💙🚀

**Pull request merged**
